### DISCUSSION TITLE
Zookeeper lose cluster information

### DISCUSSION THREAD

**Discussion by rillo-carrillo** [apache/pulsar#17286] [Q&A]
I have a zookeeper cluster on kubernetes.
Each zookeeper has it's own persistent storage.

Once zookeeper cluster is enabled I execute initialize-cluster-metadata command:
```
pulsar initialize-cluster-metadata --cluster pulsardev \
--zookeeper zookeeper-0.zookeepers:2181,zookeeper-1.zookeepers:2181,zookeeper-2.zookeepers:2181 \
--configuration-store zookeeper-0.zookeepers:2184,zookeeper-1.zookeepers:2184,zookeeper-2.zookeepers:2184 \
--web-service-url http://broker-0.brokers:8080,broker-1.brokers:8080,broker-2.brokers:8080 \
--web-service-url-tls https://broker-0.brokers:8443,broker-1.brokers:8443,broker-2.brokers:8443 \
--broker-service-url pulsar://broker-0.brokers:6650,broker-1.brokers:6650,broker-2.brokers:6650 \
--broker-service-url-tls pulsar+ssl://broker-0.brokers:6651,broker-1.brokers:6651,broker-2.brokers:6651
```
After executing I can see multiple paths created on zk
```
/
  zookeeper
  stream
  bookies
  ledgers
  pulsar
```
So far so good, but If for some reason I update some configuration on pulsar image and update happens with the RollingUpdate strategy

the paths I previously saw are "lost", now I can only see:
```
/
  admin
  zookeeper
```

Is there any way to avoid such behaviour?

**Comment by merlimat** [comment]
Is it possible that ZK servers are writing the data inside the container instead of the persistent volume?
